tests/unit/test_naming.py

```diff
@@ -31,10 +32,10 @@ def test_windows_reserved_names(self):
             'LPT5', 'LPT6', 'LPT7', 'LPT8', 'LPT9'
         }
 
-        for dirpath, dirnames, filenames in os.walk(self.esmvaltool_folder):
-            for dirname in dirnames:
-                if dirname in IGNORE:
-                    dirnames.remove(dirname)
+        for dirpath, dirnames, filenames in os.walk(self.esmvaltool_folder,
+                                                    topdown=True):
+            to_remove = [dirn for dirn in dirnames if dirn in IGNORE]
+            dirnames[:] = [dirn for dirn in dirnames if dirn not in to_remove]
```

Comment:
```suggestion
            dirnames[:] = [dirn for dirn in dirnames if dirn not in IGNORE]
```

I think that's the same, isn't it?

Could you maybe add a brief comment why this works? It's mentioned in the [doc](https://docs.python.org/3.8/library/os.html#os.walk), but I was very confused to why this should make difference.

> If optional argument topdown is True or not specified, the triple for a directory is generated before the triples for any of its subdirectories (directories are generated top-down). If topdown is False, the triple for a directory is generated after the triples for all of its subdirectories (directories are generated bottom-up). No matter the value of topdown, the list of subdirectories is retrieved before the tuples for the directory and its subdirectories are generated.

Reply:
yes! This was the tricky/key bit - so we keep only the dirs that should not be ignored and not walk on the contents of those, let me add a docstring

Reply:
and yes, that was a shimmy I put earlier to check, indeed those two lines can be collapsed now